Classes/Controller/NodeClipboard.php

```diff
@@ -0,0 +1,95 @@
+<?php
+namespace Neos\Neos\Ui\Controller;
+
+/*
+ * This file is part of the Neos.Neos.Ui package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Error\Messages\Message;
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+
+/**
+ * This is a container for clipboard state that needs to be persisted server side
+ *
+ * @Flow\Scope("session")
+ */
+class NodeClipboard
+{
+    const MODE_COPY = 'Copy';
+    const MODE_MOVE = 'Move';
+
+    /**
+     * @var NodeInterface
+     */
+    protected $node = null;
+
+    /**
+     * @var string one of the NodeClipboardInterface::MODE_*  constants
+     */
+    protected $mode = '';
```

Comment:
nitpick: if you set this to `self::MODE_COPY` here (and in `reset()`) there is one less state to consider (mode is either copy or move then)

Reply:
Make sense, but I'd leave it as `''` for the sake of how this is later treated in the UI state: would be strange if the mode is set to Copy while there's no copied node, could cause some side-effects in some of the UI logic.